src/OrbitGl/CaptureViewElement.cpp

```diff
@@ -248,4 +248,58 @@ void CaptureViewElement::RequestUpdate(RequestUpdateScope scope) {
   }
 }
 
+CaptureViewElement::RenderGroups CaptureViewElement::PreRender(
+    PrimitiveAssembler& primitive_assembler, TextRenderer& text_renderer) {
+  primitive_assembler.PushTranslation(0, 0, DetermineZOffset());
+  text_renderer.PushTranslation(0, 0, DetermineZOffset());
+
+  BatchRenderGroupStateManager* manager = primitive_assembler.GetRenderGroupManager();
+
+  RenderGroups result{primitive_assembler.GetCurrentRenderGroupName(),
+                      text_renderer.GetCurrentRenderGroupName()};
+
+  if (RequestSeparateRenderGroup()) {
+    constexpr std::string_view kGroupPrefix = "|rg_cve_";
+
+    const std::string new_batcher_render_group_name =
+        absl::StrCat(result.batcher_render_group_name, kGroupPrefix, std::to_string(GetUid()));
+    primitive_assembler.SetCurrentRenderGroupName(new_batcher_render_group_name);
+
+    BatchRenderGroupState parent_state = manager->GetGroupState(result.batcher_render_group_name);
+    BatchRenderGroupState state = parent_state;
+    state.stencil.pos = {GetPos()[0], GetPos()[1]};
+    state.stencil.size = {GetSize()[0], GetSize()[1]};
+    state.stencil.enabled = true;
+    state.stencil.ClipAt(parent_state.stencil);
+    manager->SetGroupState(new_batcher_render_group_name, state);
+
+    const std::string new_text_render_group_name =
+        absl::StrCat(result.text_render_group_name, kGroupPrefix, std::to_string(GetUid()));
+    text_renderer.SetCurrentRenderGroupName(new_text_render_group_name);
+
+    if (new_text_render_group_name != new_batcher_render_group_name) {
```

Comment:
It could happen if the parent decides to put text into a different rendering group for whatever reason. Usually this shouldn't happen unless some element explicitly wants to behave like this, but there's nothing from preventing it, so I wanted to handle this case.